### PR TITLE
Test python plugins on PR

### PR DESCRIPTION
**Pull request by Garulf**
This will test if a Python plugin that was submitted runs without errors. This will make sure a plugin will run as packaged without any extra installation.

### CODE REVIEW

**Comment by deefrawley** [comment]
I'll let one of the CI gurus approve but this will be great to have and stop broken python plugins getting into prod (especially while we get over the hump of the 'new' way to include dependencies)

**Comment by Garulf** [comment]
> I'll let one of the CI gurus approve but this will be great to have and stop broken python plugins getting into prod (especially while we get over the hump of the 'new' way to include dependencies)

Exactly my intention! Thanks for taking a look.

File: .github/workflows/test-python.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Test run Python plugins
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, edited, reopened, ready_for_review]
+    paths:
+      - 'plugins.json'
+jobs:
+  test:
```
Comment (jjw24) [inline .github/workflows/test-python.yml:9]: We ready have a ci job called test I think, so can we rename this to something else so I can also make it a requirement for merging

Comment (Garulf) [reply .github/workflows/test-python.yml]: Changed to `test-plugin`

**Comment by Garulf** [comment]
Now checks for the bottom most plugin without a `Tested` Boolean field.

**Comment by Garulf** [comment]
Okay, the python script now runs the plugin and checks it's return code. If the return code is non-zero it fails, otherwise we commit `true` to `Tested` field in the plugin manifest on in the PR.

**Comment by Garulf** [comment]
Should be ready to rock, tested numerous times.

File: .github/workflows/test-python.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Test run Python plugins
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, edited, reopened, ready_for_review]
+    paths:
+      - 'plugins.json'
+jobs:
+  test-plugin:
```
Comment (jjw24) [inline .github/workflows/test-python.yml:9]: sorry, can we change to 'test-run-plugin'? more descriptive

Comment (Garulf) [reply .github/workflows/test-python.yml]: Sure, done.

**Comment by jjw24** [comment]
> Tested field isnt updated in the manifest until the PR is accepted and merged.

To clarify, job will fail if non-zero exit code right? and will update the tested field when plugin submission is merged in?

**Comment by Garulf** [comment]
> > Tested field isnt updated in the manifest until the PR is accepted and merged.
> 
> To clarify, job will fail if non-zero exit code right? and will update the tested field when plugin submission is merged in?

Yes, if the plugin fails to run it'll return a non-zero exit and fail the test. If we accept the plugin PR it should be picked up on merge by `python-passed.yml` which adds a `Tested` field with a True` value to any python plugin missing one (which should be any plugin that got accepted).

**Review by jjw24 — APPROVED** [review]

**Comment by jjw24** [comment]
@Garulf got some failures in the ci run. Are you able to look into?

**Comment by Garulf** [comment]
> @Garulf got some failures in the ci run. Are you able to look into?

I see the issue.

**Comment by Garulf** [comment]
@jjw24 Fixed: https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest/pull/94

**Pull request merged**